### PR TITLE
Add success and failure callbacks to the directive.

### PR DESCRIPTION
**Pull request by maxpeterson**
For https://incuna-pm.atlassian.net/browse/OP-161

### CODE REVIEW

**Comment by maxpeterson** [comment]
@incuna/js please review / merge in the morning.

File: web/app/scripts/compiled-es5/directives/form.js [hunk 1]
```diff
@@ -28,37 +28,39 @@ _module.directive('surveysForm', [_api2.default.componentName, _fieldsetsParser2
         restrict: 'A',
         scope: {
             formUrl: '=',
-            responseUrl: '='
+            responseUrl: '=',
+            onSuccess: '&',
+            onFailure: '&'
         },
         templateUrl: 'templates/incuna-surveys/form/survey-form.html',
-        link: function link($scope) {
```
Comment (wytrych) [inline web/app/scripts/compiled-es5/directives/form.js]: Why do you prefer `scope` over `$scope` ?

File: web/app/scripts/compiled-es5/directives/form.js [hunk 2]
```diff
@@ -28,37 +28,39 @@ _module.directive('surveysForm', [_api2.default.componentName, _fieldsetsParser2
...
+                        scope.model = data;
                     });
                 }
             });
 
-            $scope.submit = function () {
-                if ($scope.responseUrl) {
+            scope.submit = function () {
+                if (scope.responseUrl) {
                     // TODO: Handle errors.
```
Comment (wytrych) [inline web/app/scripts/compiled-es5/directives/form.js]: Remove TODO now?

File: web/app/scripts/src/directives/form.es.js [hunk 3]
```diff
@@ -23,37 +23,41 @@ module.directive('surveysForm', [
...
+                            scope.model = data;
                         });
                     }
                 });
 
-                $scope.submit = function () {
-                    if ($scope.responseUrl) {
+                scope.submit = function () {
+                    if (scope.responseUrl) {
                         // TODO: Handle errors.
```
Comment (wytrych) [inline web/app/scripts/src/directives/form.es.js]: Remove TODO

**Comment by wytrych** [comment]
Looks, good if. You could remove TODO now, otherwise good to merge

File: web/app/scripts/src/directives/form.es.js (hunk 3 above)
Comment (maxpeterson) [inline web/app/scripts/src/directives/form.es.js]: I think we still need to handle errors properly within the form - somehow the field errors (such as this is required) need to be displayed by the fields

File: web/app/scripts/compiled-es5/directives/form.js (hunk 1 above)
Comment (maxpeterson) [inline web/app/scripts/compiled-es5/directives/form.js]: I understood `$scope` to be something that Angular injected `.controller('someCtrl' ['$scope', function ($scope) { ... }])` so it seems confusing to use it the `link` method - not sure if it any different.

Comment (wytrych) [inline web/app/scripts/compiled-es5/directives/form.js]: Fair enough

**Pull request merged**